src/main/java/io/pravega/connectors/flink/FlinkPravegaReader.java

```diff
@@ -228,7 +229,9 @@ public void run(SourceContext<T> ctx) throws Exception {
                 getRuntimeContext().getTaskNameWithSubtasks(), readerId, this.controllerURI);
 
         // create the adapter between Pravega's serializers and Flink's serializers
-        final Serializer<T> deserializer = new FlinkDeserializer<>(this.deserializationSchema);
+        final Serializer<T> deserializer = this.deserializationSchema instanceof PravegaDeserializationSchema ?
```

Comment:
Maybe instead of checking for an instance of PravegaDeserializationSchema we can introduce a new interface (that this class would implement) that implements the method:

```public Serializer<T> getSerializer()```

This way someone who is using their own serialization schema can make use of this same code if they implement that interface.